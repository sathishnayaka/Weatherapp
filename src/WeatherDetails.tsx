  import axios from 'axios';
  import React, { useState , useEffect} from 'react';
  import {
    Image,
    ImageBackground,
    SafeAreaView,
    StyleSheet,
    Text,
    useColorScheme,
  } from 'react-native';

  import {
    Colors,
  } from 'react-native/Libraries/NewAppScreen';

  type formTypes = {
    navigation : any;
    route : any;
  }

  const WeatherDetails = ({navigation,route}:formTypes) => {
    const [temp, setTemp] = useState("");
    const [weatherIcon, setWeatherIcon] = useState("");
    const [precip, setPrecip] = useState("");
    const [windSpeed, setWindSpeed] = useState("");

    const { capital } = route.params;

    const getWeatherDetails = async() => {
          const response =  await axios.get(
              `http://api.weatherstack.com/current?access_key=6f6fd10b21078247ac9030d5cebe1908&query=${capital}`
            );
          const { temperature , precip , weather_icons , wind_speed } = response.data.current;
          setTemp(temperature);
          setWeatherIcon(weather_icons[0]);
          setPrecip(precip);
          setWindSpeed(wind_speed);
          
    }

  useEffect(()=>{
      getWeatherDetails();
  },[])

    return (
      <SafeAreaView style={styles.sectionContainer}>
        <ImageBackground source={{uri : "https://cdn.dribbble.com/users/925716/screenshots/3333720/attachments/722375/night.png"}} resizeMode="cover" style={styles.image}>
        <Text data-testid="capital" style={styles.capitalText}> {capital}</Text>
        <Text data-testid="temp" style={styles.temp} > {temp}&deg;C</Text>
        <Image
          data-testid="weather-icon"
          style={styles.tinyLogo}
          source={{
            uri: weatherIcon,
          }}
        />
        <Text style={{color:'#fff', fontSize:20}}>Wind Speed : {windSpeed}KM/hr</Text>
        <Text style={{color:'#fff', fontSize:20}}> precipitation : {precip}</Text>
      </ImageBackground>
      </SafeAreaView>
    );
  }

  const styles = StyleSheet.create({
    sectionContainer: {
      width:'100%',
      height:"100%"
    },
    tinyLogo :{
      width:100, 
      height:100,
      borderRadius:16,
    },
    image :{
      flex:1,
      alignContent:'center',
      alignItems:'center',
      justifyContent:'center'
    },
    capitalText : {
      color:"#fff",
      fontSize:24,
      fontWeight:'800',
      alignSelf:'center'
    },
    temp:{
      fontSize:36,
      lineHeight:50,
      fontWeight:'800',
      color : 'orange'
    }

  });


  export default WeatherDetails;
