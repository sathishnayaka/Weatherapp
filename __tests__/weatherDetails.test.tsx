import React from 'react';
import { mount, shallow, ShallowWrapper } from 'enzyme';
import WeatherDetails from '../src/WeatherDetails';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';

describe('WeatherDetails component', () => {
  let wrapper: ShallowWrapper;
  let mockAxios: MockAdapter;

  beforeEach(() => {
    mockAxios = new MockAdapter(axios);
    mockAxios.onGet().reply(200, {
      current: {
        temperature: '25',
        weather_icons: ['http://example.com/icon.png'],
        precip: '20',
        wind_speed: '10',
      },
    });
    wrapper = shallow(<WeatherDetails navigation={{}} route={{ params: { capital: 'London' } }} />);
  });


  beforeEach(() => {
    wrapper = shallow(<WeatherDetails navigation={{}} route={{ params: { capital: 'London' } }} />);
  });
  
  afterEach(() => {
    mockAxios.restore();
  });

  it('renders the capital name correctly', () => {
    const capitalElement = wrapper.find('[data-testid="capital"]');
    expect(capitalElement.props().children[1]).toEqual('London');
  });
  it('redners temprature correclty', () => {
    const tempElement = wrapper.find('[data-testid="temp"]');
    expect(tempElement.props().children[1]).toEqual('');
  })

  it("after mounting it should call api",() => {
  
    const wrapper = mount(<WeatherDetails navigation={{}} route={{ params: { capital: 'London' } }} />);
    // console.log(wrapper.debug());
   
  })
});
