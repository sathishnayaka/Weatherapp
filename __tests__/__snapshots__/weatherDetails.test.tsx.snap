// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetails component after mounting it should call api 1`] = `ShallowWrapper {}`;
